src/main/java/com/github/manolo8/darkbot/backpage/DispatchManager.java

```diff
@@ -64,16 +65,17 @@ public Map<String, Integer> getLastCollected() {
     }
 
     public boolean hireRetriever(Retriever retriever) {
-        if (data.getAvailableSlots() <= 0) return false;
-        if (retriever.getPermitCost() > data.getPermit()) {
-            return handleResponse("Cannot hire", retriever.getId(), "(ERROR) Not enough permits");
-        }
+        if (retriever == null) return false;
         try {
+            if (data.getAvailableSlots() <= 0) return false;
+            if (retriever.getPermitCost() > data.getPermit()) {
+                return handleResponse("Hire Retriever", retriever.getId(), "(ERROR) Can Not Hire Retriever, Not enough permits");
+            }
             String response = main.backpage.getConnection("ajax/dispatch.php", Method.POST)
                     .setRawParam("command", "sendDispatch")
                     .setRawParam("dispatchId", retriever.getId())
                     .getContent();
-            return handleResponse("Hired dispatcher", retriever.getId(), response);
+            return handleResponse("Hired Dispatcher", retriever.getId(), response);
         } catch (Exception e) {
             e.printStackTrace();
             System.out.println("Exception hiring dispatcher: " + e);
```

Comment:
As mentioned in another comment, i believe it's best to show the message first then the stacktrace, apply it around the whole file, i see a few places still doing that